### PR TITLE
feat(pins): #MED-247 fetch pin resources and display them on home carousel

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

fetch pin resources and display them on home page on a carousel

image

## Checklist tests

* check all api request
* check the display with/without pin resources
* check the responsive

## Issue ticket number and link

[MED-247](https://jira.support-ent.fr/browse/MED-247)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "@emotion/react": "11.11.1",
     "@emotion/styled": "11.11.0",
     "@reduxjs/toolkit": "^2.0.1",
+    "@splidejs/react-splide": "^0.7.8",
```
Comment (niko-amosse) [inline frontend/package.json:57]: add this dependance to use carousel from splidejs library

File: frontend/public/img/no-image-resource.png
Comment (niko-amosse) [inline frontend/public/img/no-image-resource.png:1]: new image when there is no image for a resource

File: frontend/src/components/home-lists/HomeList.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ interface HomeListProps {
   handleAddFavorite: (resource: any) => void;
   handleRemoveFavorite: (id: string | number) => void;
   double?: boolean;
+  pinsEmpty: boolean;
```
Comment (niko-amosse) [inline frontend/src/components/home-lists/HomeList.tsx:29]: cette prop sert pour l'affichage du nombre de ressources sur la page d'accueil en fonction de la présence du carousel

File: frontend/src/components/home-lists/HomeList.tsx [hunk 3]
```diff
@@ -83,7 +87,10 @@ export const HomeList: React.FC<HomeListProps> = ({
             <Resource
               id={resource?.id ?? ""}
               key={resource.id}
-              image={resource?.image ?? "/mediacentre/public/img/no-avatar.svg"}
+              image={
+                resource?.image ??
+                "/mediacentre/public/img/no-image-resource.png"
+              }
```
Comment (niko-amosse) [inline frontend/src/components/home-lists/HomeList.tsx:93]: on change l'image de l'empty state

File: frontend/src/config/index.ts [hunk 4]
```diff
@@ -14,3 +14,7 @@ export const rights = {
     right: "fr-cgi-todoapp-controllers-TodoappController|delete",
   },
 };
+
+export const hatStructures = {
+  crna: "8df49fdb-faf4-4f7f-8425-414a6d5f25ff",
+};
```
Comment (niko-amosse) [inline frontend/src/config/index.ts:20]: pas encore utilisé

File: frontend/src/components/home-lists/HomeList.tsx (hunk 2 above)
Comment (ThomasSevagen) [reply frontend/src/components/home-lists/HomeList.tsx:29]: Pour un bool la convention c'est isPinsEmpty ou has....
pareil pour la prop double au dessus -> isDouble

File: frontend/src/routes/app/index.tsx [hunk 5]
```diff
@@ -45,12 +47,22 @@ export const App = () => {
   const { externalResources, setExternalResources, refetchSearch } =
     useExternalResource();
   const { globals } = useGlobal();
+  const { pins } = usePins(
+    (user?.structures && user.structures.length > 0
```
Comment (ThomasSevagen) [inline frontend/src/routes/app/index.tsx:51]: `user?.structures.length` suffit

File: frontend/src/services/api/pins.service.ts
Comment (Mattox0) [inline frontend/src/services/api/pins.service.ts:1]: appelle le service pin.service.ts, tous nos services sont au singulier

File: frontend/src/routes/app/index.tsx [hunk 6]
```diff
@@ -257,17 +269,22 @@ export const App = () => {
         </Alert>
       )}
       <div className="med-container">
-        <div className="med-fav-container">
+        <div id="pinID">{!pinsEmpty && <PinsCarousel pins={pins} />}</div>
+        <div id="favoriteID">
```
Comment (Mattox0) [inline frontend/src/routes/app/index.tsx:273]: pinId et favoriteId, conventions de nommage

File: frontend/src/routes/app/index.tsx (hunk 5 above)
Comment (niko-amosse) [reply frontend/src/routes/app/index.tsx]: fait

File: frontend/src/services/api/pins.service.ts
Comment (niko-amosse) [reply frontend/src/services/api/pins.service.ts:1]: fait

File: frontend/src/routes/app/index.tsx (hunk 6 above)
Comment (niko-amosse) [reply frontend/src/routes/app/index.tsx]: fait

**Review by Mattox0 — APPROVED** [review]

**Pull request merged**